server/graphql/mutations/orders.js

```diff
@@ -14,240 +14,230 @@ import status from '../../constants/order_status';
 import * as errors from '../errors';
 import activities from '../../constants/activities';
 import { getNextChargeAndPeriodStartDates, getChargeRetryCount} from '../../lib/subscriptions';
+import { defaultTiers } from '../../models/Collective';
 
 const debugOrder = debug('order');
 
-export function createOrder(order, loaders, remoteUser) {
-  let tier, collective, fromCollective, paymentRequired, interval, orderCreated, user;
-
-  if (order.paymentMethod && order.paymentMethod.service === 'stripe' && order.paymentMethod.uuid && !remoteUser) {
-    throw new Error("You need to be logged in to be able to use a payment method on file");
-  }
+export async function createOrder(order, loaders, remoteUser) {
+  try {
+    if (order.paymentMethod && order.paymentMethod.service === 'stripe' && order.paymentMethod.uuid && !remoteUser) {
+      throw new Error("You need to be logged in to be able to use a payment method on file");
+    }
 
-  if (!order.collective || !order.collective.id) {
-    throw new Error("No collective id provided");
-  }
+    if (!order.collective || (!order.collective.id && !order.collective.website)) {
+      throw new Error("No collective id or website provided");
+    }
 
-  if (order.platformFeePercent && !remoteUser.isRoot()) {
-    throw new Error(`Only a root can change the platformFeePercent`);
-  }
+    if (order.platformFeePercent && !remoteUser.isRoot()) {
+      throw new Error(`Only a root can change the platformFeePercent`);
+    }
 
-  // Check the existence of the recipient Collective
-  return loaders.collective.findById.load(order.collective.id)
-  .then(c => {
-    if (!c) {
-      throw new Error(`No collective found with id: ${order.collective.id}`);
+    // Check the existence of the recipient Collective
+    let collective;
+    if (order.collective.id) {
+      collective = await loaders.collective.findById.load(order.collective.id);
+    } else if (order.collective.website) {
+      collective = (await models.Collective.findOrCreate({ where: {website: order.collective.website}, defaults: order.collective }))[0];
+      collective.tiers = defaultTiers(collective.HostCollectiveId, collective.currency);
     }
-    order.collective = collective = c;
 
-    if (!collective.isActive) {
-      throw new Error(`This collective is not active`);
+    if (!collective) {
+      throw new Error(`No collective found: ${order.collective.id || order.collective.website}`);
```

Comment:
Not sure about this error message. Since you do a `findOrCreate`, the error should be "can't create a collective" no?

Reply:
If we get to this point in the code, it's because no collective was found if `order.collective.id` is passed to the mutation. 

You are correct that it will default to creating a new collective if `order.collective.website` is passed to the mutation and not found in the database.